cylc/flow/task_pool.py

```diff
@@ -1619,10 +1641,19 @@ def spawn_task(
         if not self.can_be_spawned(name, point):
             return None
 
-        submit_num, prev_status, prev_flow_wait = (
+        never_spawned, submit_num, prev_status, prev_flow_wait = (
             self._get_task_history(name, point, flow_nums)
         )
 
+        if (
+            not never_spawned and
+            not prev_flow_wait and
+            submit_num == 0
+        ):
+            # Previous spawn suicided before completing any outputs.
```

Comment:
"Never spawned" doesn't necessarily mean "suicided", I think this PR has implications on `cylc remove` too. Presumably, this will prevent tasks being reincarnated after being manually removed (as well as after being automatically suicided)?

This might be a positive behaviour change for `cylc remove`, but worth noting that is may clash with #5643.

Reply:
(Follow-up on main thread)